### DISCUSSION TITLE
This is really slick

### DISCUSSION THREAD

**Discussion by Ken-Kauffmann** [arnaud-j4k08/PicoClockGreenEasy#2] [General]
New to github, so sorry if this is not the appropriate place to comment.  I think that this implementation is amazing.  The amount of effort in writing this from scratch must have been incredible.  The user interface is much better than the example from Waveshare.  I am quite inexperienced in coding and Visual Studio Code, and I have been able to get the WIFI, and NTP to work, as well as make some very minor changes to display the day and month in US format.  The documentation is well-done, and complete, and much better than that from the Waveshare wifi.  Well done!  Thanks, Ken

**Comment by arnaud-j4k08** [comment]
Hello Ken,

This is definitely an appropriate place to compliment this project :-). I am a software engineer and I wrote this firmware during my free time, mainly to learn microcontroller programming on the Pico. The example from Waveshare did not seem to be a good base, and once I had understood how to drive the LED matrix controller, I was able to proceed with the implementation. Actually, in many cases, I took the opposite approach of Waveshare. Switching the display to DMA/PIO later was even more fun, and provides additional stability. The display even remains if the program crashes :-).

I plan to continue extending this firmware for a while. Feel free to share your ideas and experience with using it from am another country (I'm in France).

Thank you, Arnaud

**Reply by Ken-Kauffmann** [reply]
For Your Information.  I wrote a review on Amazon regarding the PicoClockGreen hardware a few weeks ago, and that review was finally accepted.  It included the following:

"For those that are comfortable with Visual Studio Code, or want to easily add additional functionality, there is an alternative implementation available. Search for PicoClockGreenEasy, for a github page and video. The source code on the github is not based on the Waveshare example, but supports all of the described functionality. IMHO, it is a pretty impressive implementation. There is a pre-built uf2 file on github, which can be installed using the normal BootSel process. The implementation supports Wifi and NTP synchronization, but compilation is required to implement those functions. The author is investigating the use of GPS for time synchronization. The use of the clock, and the code itself are well-documented, and I have already been able to customize it."

The Amazon reviewers would not accept a hyperlink, so interested customers will need to do a search.   My full review is on Amazon.

**Reply by arnaud-j4k08** [reply]
Wow! Thank you so much for praising this project on Amazon! That's a great idea. 

(I need to make a few investigations, then I'll answer your post about "Sensorless Weather Station" too)
